src/MoBi.Presentation/Tasks/SimulationCommitTask.cs

```diff
@@ -0,0 +1,126 @@
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Utility.Extensions;
+using static MoBi.Assets.AppConstants.Commands;
+
+namespace MoBi.Presentation.Tasks
+{
+   public interface ISimulationCommitTask
+   {
+      IMoBiCommand CommitSimulationChanges(IMoBiSimulation simulationWithChanges);
+   }
+
+   public class SimulationCommitTask : ISimulationCommitTask
+   {
+      private readonly IMoBiContext _context;
+      private readonly ITemplateResolverTask _templateResolverTask;
+      private readonly IEntitiesInSimulationRetriever _entitiesInSimulationRetriever;
+      private readonly IInitialConditionsCreator _initialConditionsCreator;
+      private readonly IParameterValuesCreator _parameterValuesCreator;
+      private readonly INameCorrector _nameCorrector;
+
+      public SimulationCommitTask(IMoBiContext context,
+         ITemplateResolverTask templateResolverTask,
+         IEntitiesInSimulationRetriever entitiesInSimulationRetriever,
+         IInitialConditionsCreator initialConditionsCreator,
+         IParameterValuesCreator parameterValuesCreator,
+         INameCorrector nameCorrector)
+      {
+         _context = context;
+         _templateResolverTask = templateResolverTask;
+         _entitiesInSimulationRetriever = entitiesInSimulationRetriever;
+         _initialConditionsCreator = initialConditionsCreator;
+         _parameterValuesCreator = parameterValuesCreator;
+         _nameCorrector = nameCorrector;
+      }
+
+      public IMoBiCommand CommitSimulationChanges(IMoBiSimulation simulationWithChanges)
+      {
+         var lastModuleConfiguration = simulationWithChanges.Configuration.ModuleConfigurations.Last();
+         var templateModule = _templateResolverTask.TemplateModuleFor(lastModuleConfiguration.Module);
+         
+         var moBiMacroCommand = new MoBiMacroCommand
+         {
+            CommandType = CommitCommand,
+            Description = CommitCommandDescription(templateModule, simulationWithChanges),
+            ObjectType = new ObjectTypeResolver().TypeFor<Module>()
+         };
+         moBiMacroCommand.Add(lastModuleConfiguration.SelectedInitialConditions == null ? 
+            addNewInitialConditionsFromSimulationChanges(simulationWithChanges, templateModule) : 
+            updateInitialConditionsFromSimulationChanges(simulationWithChanges, lastModuleConfiguration.SelectedInitialConditions));
+
+         moBiMacroCommand.Add(lastModuleConfiguration.SelectedParameterValues == null ? 
+            addNewParameterValuesFromSimulationChanges(simulationWithChanges, templateModule) : 
+            updateParameterValuesFromSimulationChanges(simulationWithChanges, lastModuleConfiguration.SelectedParameterValues));
+
+         return moBiMacroCommand.Run(_context);
+      }
+
+      private ICommand updateParameterValuesFromSimulationChanges(IMoBiSimulation simulationWithChanges, ParameterValuesBuildingBlock selectedParameterValues)
+      {
+         return new MoBiEmptyCommand();
```

Comment:
Throw a not implemented yet exception instead ?

Reply:
That's going to be implemented tomorrow

Reply:
and if you throw an exception, you can't test anything.